packages/go/analysis/hybrid/hybrid.go

```diff
@@ -175,8 +175,24 @@ func createMissingADUser(ctx context.Context, db graph.Database, objectID string
 	})
 
 	err = db.WriteTransaction(ctx, func(tx graph.Transaction) error {
+		newNode, err = analysis.FetchNodeByObjectID(tx, objectID)
+		if !errors.Is(err, graph.ErrNoResultsFound) && err != nil {
+			return fmt.Errorf("create missing ad user precheck: %w", err)
+		} else if err == nil {
+			newNode.AddKinds(adSchema.User)
+			if err := tx.UpdateNode(newNode); err != nil {
+				return fmt.Errorf("update missing ad user label: %w", err)
+			} else {
+				return nil
+			}
+		}
+
 		newNode, err = tx.CreateNode(properties, adSchema.Entity, adSchema.User)
-		return err
+		if err != nil {
+			return fmt.Errorf("create missing ad user: %w", err)
+		} else {
+			return nil
+		}
```

Comment:
Just a style nit per convention
```suggestion
		if newNode, err = tx.CreateNode(properties, adSchema.Entity, adSchema.User); err != nil {
			return fmt.Errorf("create missing ad user: %w", err)
		}
		
		return nil
```